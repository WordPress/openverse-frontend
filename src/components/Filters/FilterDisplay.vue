<template>
  <div class="flex flex-wrap items-center p-4" aria-live="polite">
    <span v-if="isAnyFilterApplied" class="mr-2 font-semibold mb-2">
      {{ $t('filters.filter-by') }}
    </span>
    <FilterTag
      v-for="filter in appliedFilterTags"
      :key="filter.code"
      class="mx-1 mb-2 ml-2"
      :code="filter.code"
      :label="filter.name"
      :filter-type="filter.filterType"
      @filterChanged="onUpdateFilter"
    />
  </div>
</template>

<script>
import { TOGGLE_FILTER } from '~/constants/action-types'
import FilterTag from '~/components/Filters/FilterTag'
import { mapGetters } from 'vuex'
import { FILTER } from '~/constants/store-modules'

export default {
  name: 'FilterDisplay',
  components: { FilterTag },
  computed: {
    ...mapGetters(FILTER, ['appliedFilterTags', 'isAnyFilterApplied']),
  },
  methods: {
    onUpdateFilter({ code, filterType }) {
      this.$store.dispatch(`${FILTER}/${TOGGLE_FILTER}`, { code, filterType })
    },
  },
}
</script>
