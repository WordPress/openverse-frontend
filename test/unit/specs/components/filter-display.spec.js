import { render } from '@testing-library/vue'

import FilterDisplay from '~/components/Filters/FilterDisplay'
import { createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'

describe('FilterDisplay', () => {
  const localVue = createLocalVue()
  localVue.use(Vuex)
  const storeMock = new Vuex.Store({
    modules: {
      search: {
        namespaced: true,
        state: {
          query: {
            q: 'foo',
          },
        },
      },
      filter: {
        namespaced: true,
        getters: {
          isAnyFilterApplied: () => true,
          appliedFilterTags: () => [
            {
              code: 'cc0',
              filterType: 'license',
              name: 'filters.licenses.cc0',
            },
          ],
        },
      },
    },
  })
  const options = {
    localVue,
    store: storeMock,
  }

  it('should render correct contents', () => {
    const { container } = render(FilterDisplay, options)
    expect(container.firstChild).toMatchSnapshot()
  })
})
