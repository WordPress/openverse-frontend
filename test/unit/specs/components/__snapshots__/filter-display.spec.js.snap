// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterDisplay should render correct contents 1`] = `
<div
  aria-live="polite"
  class="flex flex-wrap items-center px-6"
>
  <span
    class="me-2 font-semibold mb-2"
  >
    
    filters.filter-by
  
  </span>
   
  <button
    aria-label="filters.aria.remove-filter"
    class="button tiny tag mx-1 mb-2 ms-2"
  >
    <span>
      filters.licenses.cc0
    </span>
     
    <span
      class="close ms-2 p-2"
    >
      <i
        class="icon cross"
      />
    </span>
  </button>
</div>
`;
